quiche/src/recovery/mod.rs

```diff
@@ -1669,6 +1669,129 @@ mod tests {
+
+        // send out first packet burst (a full initcwnd).
+        for i in 0..10 {
+            let p = Sent {
+                pkt_num: i,
+                frames: smallvec![],
+                time_sent: now,
+                time_acked: None,
+                time_lost: None,
+                size: 1200,
+                ack_eliciting: true,
+                in_flight: true,
+                delivered: 0,
+                delivered_time: now,
+                first_sent_time: now,
+                is_app_limited: false,
+                tx_in_flight: 0,
+                lost: 0,
+                has_data: true,
+                is_pmtud_probe: false,
+            };
+
+            r.on_packet_sent(
+                p,
+                packet::Epoch::Application,
+                HandshakeStatus::default(),
+                now,
+                "",
+            );
+        }
+
+        assert_eq!(r.sent_packets_len(packet::Epoch::Application), 10);
+        assert_eq!(r.bytes_in_flight(), 12000);
+        assert_eq!(r.bytes_in_flight_duration(), Duration::ZERO);
+
+        // Initial pacing rate matches the override value.
+        assert_eq!(
+            r.pacing_rate(),
+            initial_pacing_rate_hint.to_bytes_per_second()
+        );
+        assert_eq!(r.get_packet_send_time(now), now);
+
+        assert_eq!(r.cwnd(), 12000);
+        assert_eq!(r.cwnd_available(), 0);
+
+        // Wait 50ms for ACK.
+        now += Duration::from_millis(50);
+
+        let mut acked = RangeSet::default();
+        acked.insert(0..10);
+
+        assert_eq!(
+            r.on_ack_received(
+                &acked,
+                10,
+                packet::Epoch::Application,
+                HandshakeStatus::default(),
+                now,
+                None,
+                "",
+            )
+            .unwrap(),
+            OnAckReceivedOutcome {
+                lost_packets: 0,
+                lost_bytes: 0,
+                acked_bytes: 12000,
+                spurious_losses: 0,
+            }
+        );
+
+        assert_eq!(r.sent_packets_len(packet::Epoch::Application), 0);
+        assert_eq!(r.bytes_in_flight(), 0);
+        assert_eq!(r.bytes_in_flight_duration(), Duration::from_millis(50));
+        assert_eq!(r.rtt(), Duration::from_millis(50));
+
+        // Pacing rate is recalculated based on initial cwnd when the
+        // first RTT estimate is available.
+        assert_eq!(
+            r.pacing_rate(),
+            expected_pacing_with_rtt_measurement.to_bytes_per_second()
+        );
+    }
+
+    #[test]
+    fn initial_pacing_rate_override_decrease_when_rtt_measured() {
+        let bw = Bandwidth::from_bytes_and_time_delta(
+            12000,
+            Duration::from_millis(50),
+        );
+        helper_initial_pacing_rate_override_test(bw * 2.0, bw);
+    }
+
+    #[test]
+    fn initial_pacing_rate_override_does_not_increase_when_rtt_measured() {
```

Comment:
It seems like the rtt stays the same here `bw * 0.5`. Also do we need 3 test cases "same", "decrease", "increase"?

Reply:
Refactored to make this a parametrized test that covers the same, decrease and increase cases.